neqo-http3/src/server.rs

```diff
@@ -138,7 +138,7 @@ impl Http3Server {
         max_datagrams: NonZeroUsize,
     ) -> OutputBatch {
         qtrace!("[{self}] Process");
-        let out = self.server.process_multiple(dgrams, now, max_datagrams);
+        let out = self.server.process_multiple_input(dgrams, now);
```

Comment:
This is the core of this pull request.

Don't generate an output datagram on the QUIC layer early on (i.e. `process_multiple`). Instead, first handle any input datagrams (`process_multiple_input`), then flush the HTTP/3 layer down to the QUIC layer (`self.process_http3`), then produce any output datagrams (`self.server.process_multiple`).